dotCMS/src/main/java/com/dotcms/util/AssetExporterUtil.java

```diff
@@ -0,0 +1,185 @@
+
+/**
+ * Utility class for exporting assets.
+ *
+ * @author vic
+ */
+public class AssetExporterUtil {
+
+    enum EXPORT_VERSION {
+        LIVE, LIVE_WORKING;
+    }
+
+    static final String LIVE_WORKING_SQL =
+                    "select working_inode, live_inode from contentlet_version_info where working_inode > ? order by" +
+                            " working_inode limit 10000";
+    static final String[] ACCEPTED = new String[] {
+            "/0/",
+            "/1/",
+            "/2/",
+            "/3/",
+            "/4/",
+            "/5/",
+            "/6/",
+            "/7/",
+            "/a/",
+            "/b/",
+            "/c/",
+            "/d/",
+            "/e/",
+            "/f/",
+            "dotcms.sql.gz"
+    };
+    static final Lazy<File> PARENT = Lazy.of(() -> new File(ConfigUtils.getAbsoluteAssetsRootPath()));
+    static final FileFilter FILE_FILTER = pathname -> {
+        for (String dir : ACCEPTED) {
+            if ((pathname.getPath()+"/").contains(dir)) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    /**
+     * Utility private and empty constructor.
+     */
+    private AssetExporterUtil() {}
+
+    public static String exportToFile(final User user, final OutputStream outputStream) {
+        final String hostName = Try
+                .of(() -> APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), false).getHostname())
+                .getOrElse("dotcms");
+        final String fileName = StringUtils.sanitizeFileName(hostName)
+                + "_assets_"
+                + DateUtil.EXPORTING_DATE_FORMAT.format(new Date())
+                + ".tar.gz";
+        Logger.info(AssetExporterUtil.class, "User : " + user.getEmailAddress() + " downloading assets");
+
+        synchronized (AssetExporterUtil.class) {
+            try {
+                exportAssets(outputStream);
+            } catch (IOException e) {
+                e.printStackTrace();
+                throw new DotRuntimeException(e);
+            }
+        }
+
+        return fileName;
+    }
+    
+    /**
+     * Builds the tar.gz file containing the assets.
+     * 
+     * @param out output stream
+     */
+     static void exportAssets(OutputStream out) throws IOException {
+        try (TarArchiveOutputStream taos =
+                     new TarArchiveOutputStream(new GZIPOutputStream(new BufferedOutputStream(out)))) {
+            taos.setBigNumberMode(TarArchiveOutputStream.BIGNUMBER_STAR);
+            taos.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU);
+
+            FileUtil.listFilesRecursively(PARENT.get(), FILE_FILTER)
+                    .stream()
+                    .filter(File::isFile)
+                    .forEach(file -> {
+                        final String relativeFilePath = file.getPath().replace(PARENT.get().getPath(), "assets");
+                        TarArchiveEntry tarEntry = new TarArchiveEntry(file, relativeFilePath);
```

Comment:
set to final

Reply:
Add some log info in order to see what is being zipped

Reply:
Done.